### PR TITLE
Gaze manager update

### PR DESCRIPTION
**Pull request by HodgsonSDAS**
References issue https://github.com/Microsoft/HoloToolkit-Unity/issues/232
Added public bool devs can flip it utilize either built in gaze stabilizer or custom.
Added GazeStabilizer.UpdateHeadStability to GazeManagers update.
Updated RayCast Hit Calc to use the stabilized head ray from GazeStabilizer
Removed redundant using of UnityEngine.VR.WSA.
Added XML Documentation.
Minor formatting organization of fields and properties.

### CODE REVIEW

**Comment by christjt** [comment]
While this code indeed solves the issue with implementing the gaze stabilizer, I feel that it is intrusive as it forces the use of the gaze stabilizer; `[RequireComponent(typeof(GazeStabilizer))]`. It may not always be the case that a user wants to use the gaze manager without using the gaze stabilizer.  And as per issue #232, what would happen if in the future we need some other gaze filter, should we then update the gaze manager to use this as well? I think a better solution if keeping the managers as simple as possible while providing the possibility of extionsion so that the developer can himself/herself deside which capabilities that should be implemented.

**Comment by HodgsonSDAS** [comment]
I definitely agree we should keep managers simple for others to extend.

I'm not sure what you mean by gaze filter.  Using the LayerMask field it's possible to pick and choose one or many layers to raycast against. Then in your cursor, or whatever you're using to consume your raycast, you can do a more refined check for a particular raycast layer.

Although, I am trying to think of a time when you _wouldn't_ want to use the gaze stabilizer.  Could you provide some examples?

**Comment by NeerajW** [comment]
I think @christjt has a good point where as you said using the GazeStabilizer is always good but someone may come up with a different low pass filter or something similar and may want to use that instead of what's available in the toolkit.

Might I recommend putting the gameObject.AddComponent<GazeStabilizer> behind a bool? Much like the stabilization plane code enabling is behind a bool value.

**Comment by HodgsonSDAS** [comment]
Ahh, okay will do.

File: Assets/HoloToolkit/Input/Scripts/GazeManager.cs [hunk 1]
```diff
@@ -77,7 +85,10 @@ public partial class GazeManager : Singleton<GazeManager>
 
         private void Awake()
         {
-            GazeStabilization = gameObject.GetComponent<GazeStabilizer>();
+            if (UseBuiltInGazeStabilization)
+            {
+                GazeStabilization = gameObject.AddComponent<GazeStabilizer>();
```
Comment (jwittner) [inline Assets/HoloToolkit/Input/Scripts/GazeManager.cs]: What if my object already contains this component? Should we check for existence before adding another?

Comment (HodgsonSDAS) [inline Assets/HoloToolkit/Input/Scripts/GazeManager.cs]: Good call

**Pull request merged**